src/ObjectStatus/ObjectStatus.js

```diff
@@ -0,0 +1,71 @@
+import classnames from 'classnames';
+import CustomPropTypes from '../utils/CustomPropTypes/CustomPropTypes';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { OBJECT_STATUS_SIZES, OBJECT_STATUS_TYPES } from '../utils/constants';
+import 'fundamental-styles/dist/icon.css';
+import 'fundamental-styles/dist/object-status.css';
+
+/** *Object Status* is a short text that represents the semantic status of an object. It has a semantic color and an optional icon.
+ * Typically, the object status is used in the dynamic page header and as a status attribute of a line item in a table. */
+
+const ObjectStatus = React.forwardRef(({ children, className, glyph, indication, inverted, link, onClick, size, status, ...props }, ref) => {
+    const objectStatusClasses = classnames(
+        'fd-object-status',
+        {
+            [`sap-icon--${glyph}`]: !!glyph,
+            [`fd-object-status--indication-${indication}`]: !!indication,
+            'fd-object-status--inverted': inverted,
+            'fd-object-status--large': size === 'l',
+            [`fd-object-status--${status}`]: !!status,
+            ['fd-object-status--link']: !!link || !!onClick
+        },
+        className
+    );
+    const StatusTag = typeof link !== 'undefined' ? 'a' : 'span';
+
+    return (
+        <StatusTag
+            {...props}
+            className={objectStatusClasses}
+            href={link}
+            onClick={onClick}
```

Comment:
I think if it has `onClick` it will also need a `role='button'`, but feel free to correct me @prsdthkr / @meganaconley  / @jacobdevera

Reply:
If it's a `span` it should have `role='button'` and also support using `space` and `enter` to activate whatever the `onClick` is doing. If it's an `a` then only `enter` needs to be supported and it should not have `role='button'`.

Reply:
We will also need the buttons and links to have tab index.
My suggestion:
```js
    let StatusTag = 'span';
    let semanticProps = {};

    if (link?.length) {
        StatusTag = 'a';
        semanticProps.href = link;
    } else if ( typeof onClick === 'function') {
        StatusTag = 'button';
        semanticProps.onClick = onClick;
        semanticProps.style = {
            background: 'transparent'
        };
    }
    return (
        <StatusTag
            {...props}
            className={objectStatusClasses}
            {...semanticProps}
            ref={ref}>
            {children}
        </StatusTag>
    );
```